### PR TITLE
transactions - dont throw if chain id is not a string

### PR DESCRIPTION
**Pull request by frankiebee**

### CODE REVIEW

**Comment by frankiebee** [comment]
fixes #3841

File: app/scripts/lib/tx-state-manager.js [hunk 1]
```diff
@@ -140,8 +140,8 @@ module.exports = class TransactionStateManager extends EventEmitter {
   validateTxParams(txParams) {
     Object.keys(txParams).forEach((key) => {
       const value = txParams[key]
-      if (typeof value !== 'string') throw new Error(`${key}: ${value} in txParams is not a string`)
-      if (!ethUtil.isHexPrefixed(value)) throw new Error('is not hex prefixed, everything on txParams must be hex prefixed')
+      if (typeof value !== 'string' && key !== 'chainId') throw new Error(`${key}: ${value} in txParams is not a string`)
```
Comment (kumavis) [inline app/scripts/lib/tx-state-manager.js]: can we instead validate that chainId is undefined or a Number

**Review by kumavis — CHANGES_REQUESTED** [review]

**Comment by kumavis** [comment]
test-e2e failure is just a blank screen - maybe we need to allow more time for boot (?):
https://11155-42009758-gh.circle-artifacts.com/0/test-artifacts/should%20open%20Metamask.io/test-failure-screenshot.png

**Comment by kumavis** [comment]
maybe try out these
https://www.npmjs.com/package/validate
https://www.npmjs.com/package/joi

**Review by kumavis — APPROVED** [review]

**Pull request merged**